UUID.xs

```diff
@@ -357,16 +357,19 @@ PREINIT:
 CODE:
    RETVAL = (uuid_context_t *)PerlMemShared_malloc(sizeof(uuid_context_t));
    if ((fd = fopen(UUID_STATE_NV_STORE, "rb"))) {
-      fread(&(RETVAL->state), sizeof(uuid_state_t), 1, fd);
+      if (! fread(&(RETVAL->state), sizeof(uuid_state_t), 1, fd)) {
+          croak("fread failed");
+      }
       fclose(fd);
```

Comment:
calling croak(...) here leaves the fd open and leaks the file handle.

To see this in action on linux:

    mkdir /tmp/.UUID_STATE
    perl -MData::UUID -e 
    perl -Iblib/lib -Iblib/arch -MData::UUID -e 'eval { Data::UUID->new } for 1..100; <>;'

And in another terminal:

    lsof | grep .UUID_STATE

We should fclose(fd) before calling croak every time that fd is already open.

Thanks.